test-tools/src/main/java/io/neow3j/test/TestBlockchain.java

```diff
@@ -0,0 +1,157 @@
+package io.neow3j.test;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+public interface TestBlockchain {
```

Comment:
Why don't you make `TestBlockchain<T>` and then in each method that returns `TestBlockchain` (e.g., `withSecondsPerBlock`) you simply do `T withSecondsPerBlock(int secondsPerBlock);`

I believe that's possible, but you need to check. 😄

Reply:
hmmm, I don't see the rationale behind this. 
Then, if you used an instance of `TestBlockchain` as follows, you get back type `T` which isn't a `TestBlockchain` anymore, i.e., method chaining is not possible.
```
chain.withSecondsPerBlock(annotation.blockTime()). ?
```

Reply:
hmmm.. the `chain` will be a `TestBlockchain` anyway... because the class of the `chain` variable will implement the interface anyway.

Also, it will give the possibility to developers to chain other methods they would like to implement (besides the TestBlockchain interface).